bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/firmware/ModelRestrictedFirmwareUpdateServiceOSGiTest.java

```diff
@@ -219,9 +219,9 @@ private FirmwareProvider createFirmwareProvider(Firmware... firmwares) {
             @Override
             public Set<Firmware> getFirmwares(ThingTypeUID thingTypeUID, Locale locale) {
                 if (thingTypeUID.equals(thingType.getUID())) {
-                    return Sets.newHashSet(firmwares);
+                    return new HashSet<>(Arrays.asList(firmwares));
                 } else {
-                    return Sets.newHashSet();
+                    return new HashSet<>();
```

Comment:
this might also be `Collections.emptySet()`

Reply:
Right - I had a similar case where groovy was involved (and it didn't like it). But you're right in this case.